libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -501,6 +501,26 @@ void BaseDaemon::closeFDs()
     }
 }
 
+namespace {
+/// In debug version on Linux, increase oom score so that clickhouse is killed
+/// first, instead of some service.
+#if defined(__linux__) and not defined(NDEBUG)
+void debug_adjust_oom_score()
+{
+    int file = open("/proc/self/oom_score_adj", O_WRONLY);
+    if (file < 0)
+    {
+        return;
+    }
+    const char score[] = "555";
```

Comment:
No motivation for 555.

Reply:
A random number from [-1000, 1000], which is the range of oom_adjust_score. chrome uses something like 200 or 300.